### PR TITLE
use existing logic to configure optimizers in lr_finder

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

See title

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/trainer/optimizers.py [hunk 1]
```diff
@@ -40,6 +40,14 @@ def init_optimizers(self, model: Optional["pl.LightningModule"]) -> Tuple[List,
             )
             optim_conf = _MockOptimizer()
 
+        optimizers, lr_schedulers, optimizer_frequencies = self.configure_optimizers_and_schedulers(
```
Comment (carmocca) [inline pytorch_lightning/trainer/optimizers.py:43]: Can this wait until https://github.com/PyTorchLightning/pytorch-lightning/pull/9666 lands? To avoid the author getting conflicts

Comment (rohitgr7) [reply pytorch_lightning/trainer/optimizers.py:43]: yeah sure!

File: pytorch_lightning/tuner/lr_finder.py [hunk 2]
```diff
@@ -98,25 +97,16 @@ def __init__(self, mode: str, lr_min: float, lr_max: float, num_training: int):
         self.results = {}
         self._total_batch_idx = 0  # for debug purpose
 
-    def _exchange_scheduler(self, configure_optimizers: Callable):
+    def _exchange_scheduler(self, trainer: "pl.Trainer", model: "pl.LightningModule"):
         """Decorate configure_optimizers methods such that it returns the users originally specified optimizer
         together with a new scheduler that that takes care of the learning rate search."""
+        configure_optimizers = model.configure_optimizers
 
         @wraps(configure_optimizers)
```
Comment (carmocca) [inline pytorch_lightning/tuner/lr_finder.py:105]: Your PR changes look good given the existing Tuner design but it builds on top of the existing "hacks" we already have and perhaps we should stop that and start to unravel.

I think that instead of wrapping the `LightningModule` hook to inject the logic, this should patch the functions in `TrainerOptimizersMixin`, otherwise the implementations will continue to diverge

(for example, the tuner does not use `call_hook`)
